dashboard/src/actions/authActions.js

```diff
@@ -54,16 +55,17 @@ export const movePage = (toPage, navigate) => async (dispatch) => {
   navigate(toPage);
 };
 
-export const logout = () => async (dispatch, getState) => {
+export const clearCachedSession = () => async (dispatch, getState) => {
   dispatch({ type: TYPES.LOADING });
-  const keycloak = getState().apiEndpoint.keycloak;
-  const keys = ["username", "isLoggedIn"];
-  for (const key of keys) {
-    Cookies.remove(key);
-  }
-  keycloak.logout();
+  Cookies.remove("isLoggedIn");
   dispatch({ type: TYPES.COMPLETED });
   setTimeout(() => {
     window.location.href = APP_ROUTES.AUTH;
   }, CONSTANTS.LOGOUT_DELAY_MS);
 };
+
+export const sessionLogout = () => async (dispatch, getState) => {
+  const keycloak = getState().apiEndpoint.keycloak;
+  keycloak.logout();
+  dispatch(clearCachedSession());
```

Comment:
Y'know, you don't _have_ to call `dispatch()`, here, right?  This
```javascript
clearCachedSession()(dispatch, getState);
```
would do just fine.

And, I don't see any reason why `clearCachedSession()` needs to be a function which returns a function.  If you fix it to just be an `async` function, then you can get rid of the extra call/redirection (here and in the other invocations).